waiter/src/waiter/auth/authentication.clj

```diff
@@ -283,28 +286,28 @@
         {:keys [expires-at]} cookie-metadata
         ;; handle legacy cookies which will not have this value set
         expires-at (or expires-at current-epoch-time)
-        waiter-token (some-> request utils/request->host utils/authority->host)]
+        request-host (some-> request utils/request->host utils/authority->host)]
     (log/info auth-principal "cookie expires at" expires-at "offset is" offset-parsed)
     (cond
       ;; avoid infinite redirect loop
       done
       (-> {:status http-204-no-content}
         (utils/attach-waiter-source)
-        (assoc :waiter/token waiter-token))
+        (assoc :waiter/token request-host))
```

Comment:
Contexts outside of this function may reasonably believe `:waiter/token` to always contain valid (extant) Waiter tokens.  We should validate that this is, in fact, a Waiter token and return an error code if it is not.

Reply:
Added check for valid tokens.